tests/snapshot_tests/snapshot_apps/disable_widgets.py

```diff
@@ -0,0 +1,84 @@
+from textual.app import App, ComposeResult
+from textual.containers import Vertical, Horizontal
+from textual.widgets import (
+    Header,
+    Footer,
+    Button,
+    DataTable,
+    Input,
+    ListView,
+    ListItem,
+    Label,
+    Markdown,
+    MarkdownViewer,
+    Tree,
+    TextLog,
+)
+
+
+class WidgetDisableTestApp(App[None]):
+
+    CSS = """
+    Horizontal {
+        height: auto;
+    }
+    DataTable, ListView, Tree, TextLog {
+        height: 2;
+    }
+
+    Markdown, MarkdownViewer {
+        height: 1fr;
+    }
+    """
+
+    @property
+    def data_table(self) -> DataTable:
```

Comment:
This is an interesting use of `@property`! In an actual app, this would return a new `DataTable` each time you access it, right? So was this a "quick & dirty" way of separating the creation of the widgets from their `yield`ing in `compose` or am I missing something?

Reply:
> In an actual app, this would return a new DataTable each time you access it, right?

Correct. It's just a way of keeping the main `compose` tidy. Think of it more as a "macro" than a property for the purposes of generating the snapshot. It'd be a horrible idea for a proper app.